### ISSUE TITLE
No syntax error on setter with constructor initialization list

### ISSUE THREAD

**Issue by munificent** [dart-lang/sdk#41121]
If you try to run dartfmt on this code:

```dart
class ConfigurationService {
  Configuration _configuration;

  void set configuration(Configuration configuration) :
    assert(configuration != null), _configuration = configuration;
}
```

Instead of getting an error from the parser and aborting, it gets an AST back, tries to format it, and then fails when the output isn't valid. This code is definitely broken. Setters can't have initialization lists! :)

Usually bugs like this are mis-categorized compile-time errors in the parser. The formatter only cares about errors with `ErrorType.SYNTACTIC_ERROR` so other types get ignored. But in this case, it doesn't look like the parser reports any error at all as far as I can tell.

See: https://github.com/dart-lang/dart_style/issues/892

cc @bwilkerson @jensjoha

**Comment by jensjoha** [comment]
This should have been fixed with 4ab3bda5dbf75de4c0137e41b03ef5d9b5a4a94d (although that particular code crashes the frontend --- so don't try to compile it ;))

**Comment by munificent** [comment]
Yup, I can confirm this fixes dartfmt. Thanks!
